torch/_inductor/codegen/cpp_wrapper_cpu.py

```diff
@@ -259,6 +257,40 @@ def write_input_output_info(
     ):
         self.prefix.writeline(f"""{info_kind}[{idx}].name = "{name}";""")
 
+    def codegen_input_symbol_assignment(
+        self,
+        code: IndentedBuffer,
+        name: str,
+        value: ir.TensorBox,
+        bound_vars: Set[sympy.Symbol],
+    ):
+        @functools.lru_cache(None)
+        def sizeof(name):
+            self.codegen_input_size_var_decl(code, name)
+            return f"{name}_size"
+
+        @functools.lru_cache(None)
+        def strideof(name):
+            self.codegen_input_stride_var_decl(code, name)
+            return f"{name}_stride"
+
+        if isinstance(value, sympy.Expr):
+            if not isinstance(value, sympy.Symbol) or value in bound_vars:
+                return
+            code.writeline(f"int64_t {value} = {name};")
```

Comment:
nit - is this safe to always declare an ```int64_t``` here? If we have any assumption about the type of ```value``` and ```name```, we might want to assert such an assumption?

Reply:
We only expect to see symint at the moment, but I have extended it to symfloat following your suggestion.